pkg/kube/utils.go

```diff
@@ -171,3 +173,30 @@ func PVCContainsReadOnlyAccessMode(pvc *corev1.PersistentVolumeClaim) bool {
 
 	return false
 }
+
+// AddLabelsToPodOptions adds additional label selector to `PodOptions`. It extracts the value from the context
+// if targetkey is present and assigns to the podoptions.
+func AddLabelsToPodOptionsFromContext(
+	ctx context.Context,
+	options *PodOptions,
+	targetKey string,
+) {
+	fields := field.FromContext(ctx)
+	if fields == nil {
+		return
+	}
+	var targetValue = ""
+	for _, f := range fields.Fields() {
+		if f.Key() == targetKey {
+			targetValue = f.Value().(string)
+		}
+	}
+	// Not found
+	if targetValue == "" {
+		return
+	}
+	if options.Labels == nil {
+		options.Labels = make(map[string]string)
+	}
+	options.Labels[targetKey] = targetValue
```

Comment:
```suggestion
	if options.Labels == nil {
		options.Labels = make(map[string]string)
	}
	for _, f := range fields.Fields() {
		if f.Key() == targetKey {
			options.Labels[targetKey] = f.Value().(string)
			return
		}
	}
```

Reply:
With this suggestion, I am thinking even if the key is not present, the option.Labels is initialized. I did not want to do this initialization in case my targetKey is absent. wdyt ?

Reply:
Done